sdk/python/kfp/components/_components.py

```diff
@@ -365,13 +365,15 @@ def component_default_to_func_default(component_default: str, is_optional: bool)
     ],
 )
 
+def _generate_input_value(value, type_name: str, input_name: str) -> str:
+    return serialize_value(value, type_name)
 
 def _resolve_command_line_and_paths(
     component_spec: ComponentSpec,
     arguments: Mapping[str, str],
     input_path_generator=_generate_input_file_name,
     output_path_generator=_generate_output_file_name,
-    argument_serializer=serialize_value,
+    input_value_generator=_generate_input_value,
```

Comment:
This might not be needed. See https://github.com/kubeflow/pipelines/pull/4529/files#r492390917

Reply:
(also see the other reply)
The parameter name `argument_serializer` is actually misleading IMO, it only applies to `InputValuePathHolder` not the whole `arguments`. I was trying to find a quick way to make this work for v2 while not breaking v1. 
Another option I'm thinking is to fork `kfp/components` to `kfp/v2/components` then this part of the code can be rewritten in a more succinct way that only work for IR.